packages/amplify-cli/src/commands/env/checkout.js

```diff
@@ -22,12 +24,40 @@ module.exports = {
     const jsonString = JSON.stringify(localEnvInfo, null, 4);
     fs.writeFileSync(localEnvFilePath, jsonString, 'utf8');
 
-    // Initialize the environment
+
+    // Setup exeinfo
 
     context.amplify.constructExeInfo(context);
     context.exeInfo.forcePush = false;
+    context.exeInfo.isNewEnv = false;
     context.exeInfo.restoreBackend = context.parameters.options.restore;
 
+    // Setup Provider creds/info
+    const initializationTasks = [];
+    const providerPlugins = getProviderPlugins(context);
+    context.exeInfo.projectConfig.providers.forEach((provider) => {
+      const providerModule = require(providerPlugins[provider]);
+      initializationTasks.push(() => providerModule.init(
+        context,
+        allEnvs[envName][provider],
+      ));
+    });
+
+    await sequential(initializationTasks);
+
+    const onInitSuccessfulTasks = [];
+    context.exeInfo.projectConfig.providers.forEach((provider) => {
+      const providerModule = require(providerPlugins[provider]);
+      onInitSuccessfulTasks.push(() => providerModule.onInitSuccessful(
```

Comment:
may be name it `afterInit`

Reply:
How is this related to codegen headless?

Reply:
Take a look at the PR heading - fix: codegen headless fix & ask AWS profile info when info not present when checking out a new env

Reply:
Also onInitSuccessful is a function defined in providers and it’s used in multiple places - also in the provider from the beginning - not something introduced in this PR

Reply:
May be these should be 2 sepearate PRs then. Just so that we can create the right change log